persistence/jdbc/src/main/java/org/infinispan/persistence/jdbc/stringbased/JdbcStringBasedStore.java

```diff
@@ -466,18 +504,30 @@ public boolean contains(Object key) {
       return load(key) != null;
    }
 
-   private <P> Flowable<P> publish(Function<ResultSet, Flowable<P>> function) {
+   private <P> Flowable<P> publish(IntSet segments, Function<ResultSet, Flowable<P>> function) {
       return Flowable.using(() -> {
          Connection connection = connectionFactory.getConnection();
-         String sql = tableManager.getLoadNonExpiredAllRowsSql();
+         String sql;
+         if (segments != null) {
+            // TODO: need to handle segments when not segmented
```

Comment:
What scenario is this?

Reply:
I actually just clarified this in the API with the most recent commit. I need to revisit this code to make sure it is okay. Basically single key operations (load,write, delete) can invoke either method (segmented or not) where as the bulk ones can only invoke the segment method when it is segmented.

Reply:
This TODO can be removed. The segments won't be provided when the store is segmented, it is done at the PersistenceManagerImpl level instead.

https://github.com/infinispan/infinispan/blob/master/core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java#L565